### PR TITLE
Clean pod schedule annotation

### LINKED ISSUE #268: clean pod-scheduling annotation when the last pod was schedueld
The PVC annotation [`AnnPVCPodScheduling `](https://github.com/pingcap/tidb-operator/pull/266/files#diff-5e9724242fffab7ce692f8161cb209bcR50) is used to represent whether the pod is scheduling, it should be cleaned when the last pod was scheduled.

#266

### PR DESCRIPTION
**Pull request by onlymellb**
### What problem does this PR solve? 
fix #268

### Check List 

Tests 

 - Unit test
-  E2E test
 - Manual test 

Code changes

 - Has Go code change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:
 
 ```release-note
None
 ```

### CODE REVIEW

File: pkg/manager/member/pvc_cleaner.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		pvcName := pvc.GetName()
+		l := label.Label(pvc.Labels)
+		if !(l.IsPD() || l.IsTiKV()) {
+			skipReason[pvcName] = skipReasonPVCCleanerIsNotPDOrTiKV
+			continue
+		}
+
+		if pvc.Annotations[label.AnnPVCDeferDeleting] != "" {
+			if _, exist := pvc.Annotations[label.AnnPVCPodScheduling]; !exist {
+				// The defer deleting PVC has not pod scheduling annotation, do nothing
```
Comment (weekface) [inline pkg/manager/member/pvc_cleaner.go]:
```suggestion
				// The defer deleting PVC has no pod scheduling annotation, do nothing
```

File: pkg/scheduler/predicates/ha.go [hunk 2]
```diff
@@ -302,7 +304,31 @@ func (h *ha) realPVCListFn(ns, instanceName, component string) (*apiv1.Persisten
...
 
 func (h *ha) realUpdatePVCFn(pvc *apiv1.PersistentVolumeClaim) error {
-	_, err := h.kubeCli.CoreV1().PersistentVolumeClaims(pvc.GetNamespace()).Update(pvc)
+	ns := pvc.GetNamespace()
+	tcName := pvc.Labels[label.InstanceLabelKey]
+	pvcName := pvc.GetName()
+
+	labels := pvc.GetLabels()
+	ann := pvc.GetAnnotations()
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```
Comment (weekface) [inline pkg/scheduler/predicates/ha.go]:
```suggestion
	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

**Comment by tennix** [comment]
This PR does not have user facing changes, so I removed the release note.

**Pull request merged**